src/Test.FunctionalTests.BinSkim.Rules/RuleTests.cs

```diff
@@ -1099,6 +1126,15 @@ public void BA2025_EnableShadowStack_Fail()
             }
         }
 
+        [Fact]
+        public void BA2025_EnableShadowStack_NotApplicable()
+        {
+            this.VerifyApplicability(
+                skimmer: new EnableShadowStack(),
+                applicabilityConditions: new HashSet<string>(),
```

Comment:
>new HashSet<string>() [](http://example.com/codeflow?start=41&length=21)

In general, I think requiring that a caller instantiate an object which is empty indicates a design that could be improved, i.e., passing 'null' here may be a preferable indicator of no applicability conditions. 

Make sense? This is a fairly subtle API consideration, not a big deal. Sometimes it is important to preserve null for some specific scenario, as distinct from a non-null object which is empty.